### PR TITLE
refactor: move Carousel to css in js

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  move Carousel to css in js         |
| 🇨🇳 Chinese |  move Carousel to css in js         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/carousel/index.tsx [hunk 1]
```diff
@@ -78,12 +79,18 @@ const Carousel = React.forwardRef<CarouselRef, CarouselProps>(
       typeof dots === 'boolean' ? false : dots?.className,
     );
 
-    const className = classNames(prefixCls, {
-      [`${prefixCls}-rtl`]: direction === 'rtl',
-      [`${prefixCls}-vertical`]: newProps.vertical,
-    });
+    const [wrapSSR, hashId] = useStyle({ prefixCls, antPrefixCls: getPrefixCls() });
```
Comment (zombieJ) [inline components/carousel/index.tsx:82]: 最好还是 `useStyle(prefixCls, rootPrefixCls)`，object 的 key 无法被 uglify 掉

File: components/carousel/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,367 @@
...
+
+      '.slick-track': {
+        position: 'relative',
+        top: 0,
+        insetInlineStart: 0,
+        display: 'block',
+
+        '&::before, &::after': {
+          display: 'table',
+          content: '',
```
Comment (zombieJ) [inline components/carousel/style/index.tsx:89]: `'""'`

File: components/carousel/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,367 @@
...
+        '.slick-loading &': {
+          visibility: 'hidden',
+        },
+      },
+
+      '.slick-slide': {
+        display: 'none',
+        float: 'left',
+        height: '100%',
+        minHeight: '1px',
```
Comment (zombieJ) [inline components/carousel/style/index.tsx:105]: `1`

File: components/carousel/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,367 @@
...
+        '&::before, &::after': {
+          display: 'table',
+          content: '',
+        },
+
+        '&::after': {
+          clear: 'both',
+        },
+
+        '.slick-loading &': {
```
Comment (zombieJ) [inline components/carousel/style/index.tsx:96]: 这个样式看起来会把 hash 带进去

File: components/carousel/style/index.tsx [hunk 5]
```diff
@@ -1,2 +1,367 @@
...
+      '.slick-arrow.slick-hidden': {
+        display: 'none',
+      },
+
+      // Arrows
+      '.slick-prev, .slick-next': {
+        position: 'absolute',
+        top: '50%',
+        display: 'block',
+        width: '20px',
```
Comment (zombieJ) [inline components/carousel/style/index.tsx:142]: =>  `20`，有直接写的数字加个 `// FIXME hardcode in v4`，下同

File: components/carousel/style/index.tsx [hunk 6]
```diff
@@ -1,2 +1,367 @@
...
+            padding: 0,
+            color: 'transparent',
+            fontSize: 0,
+            background: token.colorBgComponent,
+            border: 0,
+            borderRadius: '1px',
+            outline: 'none',
+            cursor: 'pointer',
+            opacity: 0.3,
+            transition: 'all 0.5s',
```
Comment (zombieJ) [inline components/carousel/style/index.tsx:237]: 这个可以用 durationSlow 的 0.3s 代替么？

File: components/carousel/style/index.tsx (hunk 2 above)
Comment (afc163) [reply components/carousel/style/index.tsx:89]: 哦？我还特意改成这样。

File: components/carousel/style/index.tsx (hunk 6 above)
Comment (afc163) [reply components/carousel/style/index.tsx:237]: 感觉可以

File: components/carousel/style/index.tsx (hunk 4 above)
Comment (afc163) [reply components/carousel/style/index.tsx:96]: 确实。。

图片

Comment (afc163) [reply components/carousel/style/index.tsx:96]: 明天再看

**Pull request merged**